test/tic_tac_toe_test.exs

```diff
@@ -2,7 +2,32 @@ defmodule TicTacToeTest do
   use ExUnit.Case
   import ExUnit.CaptureIO
 
-  test "displays the welcome message" do
-    assert capture_io(fn -> TicTacToe.welcome_message() end) === "Welcome to Tic-Tac-Toe"
+
+  describe "start/1" do
+    test "displays welcome message and board" do
+      assert capture_io(fn -> TicTacToe.start(CommandLine) end) ===
+        """
+        Welcome to Tic-Tac-Toe
+
+        1 | 2 | 3
+        --|---|--
+        4 | 5 | 6
+        --|---|--
+        7 | 8 | 9
+        """
+    end
+
+    test "displays welcome message and board using mock" do
+      assert TicTacToe.start(CommandLineMock) ===
```

Comment:
If we're treating `CommandLineMock` as just another `writer` maybe we should just call it `WriterMock` or something like that.

Reply:
This is really good point that I did not see before but completely understand and agree with. Since the `CommandLineMock` is simply just returning the message it receives (super generic in terms of working for any I/O), I can for sure see `WriterMock` or something more generic being a better name.

Reply:
I was thinking about using the term `Communicator` or something in the future, (I like the name `Collaborator` as well), but for now until the functionality changes from solely writing to the command line (or whichever we are using to display the results) I think it makes the most sense to keep it as `writer`, but I will for sure keep this in mind for the future! Thank you 🙏